errors.go

```diff
@@ -319,3 +318,58 @@ func Find(origErr error, test func(error) bool) error {
 	})
 	return foundErr
 }
+
+// Chain ties together two errors, giving them a Causer interface and a concatenated Error message.
+type chain struct {
+	error
+	prevErr error
+}
+
+func (err *chain) Error() string {
+	return err.error.Error()
+}
+
+func (err *chain) Cause() error {
+	return err.prevErr
```

Comment:
hi @gregwebs , I check juju's impl again. It seem will return `.error` in `Cause()`, and `Error()` will go `preErr` when `error == nil`...

expect this two question, others are look good to me